src/encoded/static/components/matrix_deeply_profiled_uniform_batch.js

```diff
@@ -209,7 +209,12 @@ const convertDeeplyProfileDatatToDataTable = (context, getRowCategories, getRowS
     let matrixRow = 1;
     const matrixDataTable = rowCategoryData.sort((a, b) => sortByDescending(a.key, b.key)).reduce((accumulatingTable, rowCategoryBucket, rowCategoryIndex) => {
         const subCategoryData = getRowSubCategories(rowCategoryBucket);
-        const rowCategoryColor = rowCategoryColors[rowCategoryIndex];
+
+        // ensure colorIndex does not exceed the size of rowCategoryColors.
+        const rowCategoryColorsLength = rowCategoryColors.length;
+        const colorIndex = rowCategoryIndex < rowCategoryColorsLength ? rowCategoryIndex : (rowCategoryIndex % rowCategoryColorsLength);
```

Comment:
To be pedantic, this is equivalent to:
```
const colorIndex = rowCategoryIndex % rowCategoryColorsLength;
```
In which case you _can_ slip that expression between the brackets of line 217. That’s a pretty common paradigm: `a = b[c % b.length]`